static/js/src/entry/account/routes/membership/components/RecurringOrCircle.vue

```diff
@@ -1,13 +1,37 @@
 <template>
   <section class="c-detail-box">
     <div class="has-xxl-btm-marg">
+      <p
+        v-if="declinedCard"
+        role="alert"
+        class="has-b-btm-marg has-text-error"
+      >
+        <strong>Card was declined</strong>
+      </p>
+      <p
+        v-if="successMessage"
+        role="alert"
+        class="has-b-btm-marg has-text-success"
+      >
+        <strong>{{ successMessage }}</strong>
+      </p>
       <info-list :items="data">
         <template #text="{ item: { extra, key } }">
           <template v-if="key === 'donation'">
             {{ extra.amount | currency }}, {{ extra.period }}
           </template>
           <template v-if="key === 'payment'">
             {{ extra.brand }} ending in {{ extra.last4 }}
+            <button @click="openPaymentForm = !openPaymentForm">
+              <icon v-if="!openPaymentForm" name="pencil-fill" :display="{ size: 's' }" />
+              <icon v-if="openPaymentForm" name="close" :display="{ size: 's' }" />
+            </button>
+            <card-update
+              v-if="openPaymentForm"
+              @formSubmitted="formSubmitted"
+              @onSuccess="onSuccess"
+              @badCard="badCard"
+            ></card-update>
```

Comment:
Style-wise, I think the pencil icon looks slick, I like how it transforms into an X on click. But I have two concerns:

A) I feel it looks good when it's on the same line, but looks a bit weirder when it falls below it. I think if we go with the icon, it might be best to have it all on one line.

![Screen Shot 2022-10-11 at 11 58 12 AM](https://user-images.githubusercontent.com/37225902/195157905-2c74af4b-789e-4131-8045-946179cb737d.png)

B) Maybe it's just me, but part of me worries more generally that clicking on the pencil icon to edit credit card details may not be intuitive to older or less tech-saavy users. I'd hate to lose the icon, but I do wonder if this is a case where we need to be a bit more explicit. We could replace the icon with text or a button that simply reads "Edit" or "Edit details". A compromise might to include "Edit" with the icon. For example, something like:

![Screen Shot 2022-10-11 at 12 18 16 PM](https://user-images.githubusercontent.com/37225902/195158117-1757a9e7-732d-4a9f-b703-3036ee6f6e8a.png)

Reply:
+1! I think the word + icon is most intuitive. Great idea @SimmonsRitchie

Reply:
Yeah, good call on this @SimmonsRitchie. Done and done.